jdaviz/app.py

```diff
@@ -402,11 +403,20 @@ def _link_new_data(self, reference_data=None, data_to_be_linked=None):
         ref_data = dc[reference_data] if reference_data else dc[0]
         linked_data = dc[data_to_be_linked] if data_to_be_linked else dc[-1]
 
+        # The glue-astronomy SpectralCoordinates currently seems incompatible with glue
+        # WCSLink. This gets around it until there's an upstream fix.
+        if isinstance(linked_data.coords, SpectralCoordinates):
+            wc_old = ref_data.world_component_ids[0]
+            wc_new = linked_data.world_component_ids[0]
+            self.data_collection.add_link(LinkSame(wc_old, wc_new))
+            return
+
         try:
             if linked_data.meta.get("Plugin", None) == 'GaussianSmooth':
                 raise AttributeError
             dc.add_link(WCSLink(ref_data, linked_data))
         except (AttributeError, IncompatibleWCS):
+            raise
```

Comment:
is this intentional or for debugging purposes?

Reply:
Whoops, debugging purposes, good catch! Will fix shortly.